keps/sig-auth/20200507-external-tls-certificate-authenticator/README.md

```diff
@@ -0,0 +1,749 @@
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI, CRI
+  or CNI may require updating that component before the kubelet.
+  -->
+
+## Implementation History
+
+<!--
+Major milestones in the life cycle of a KEP should be tracked in this section.
+Major milestones might include
+- the `Summary` and `Motivation` sections being merged signaling SIG acceptance
+- the `Proposal` section being merged signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+  -->
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+## Alternatives
+
+<!--
+What other approaches did you consider and why did you rule them out?  These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+### External signer vs existing authenticators using TLS certificates
+
+First of all, it should be noted that Kubernetes already offers authentication
+using client certificates, enabled by passing the `--client-ca-file=SOMEFILE`
+option to API server, or by using the client-go [credential
+plugins](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins)
+(`exec`), which executes an arbitrary command that returns
+`clientCertificateData` and `clientKeyData`. However, both of these alternatives
+require kubectl to obtain a direct access to the client key data, what can be
+considered unacceptable in some regulated environments and therefore does not
+fullfil our requirements.
+
+### Monolithic vs modular architecture
+
+An alternative approach to enable HSM support within kubectl would be to create
+a monolithic authentication provider, which implements the PKCS#11 protocol. The
+drawback of this approach is that the support of PKCS#11 by definition requires
+making C calls to the .so library, which makes CGO mandatory during the build
+process of the whole kubectl. Such a dependency has been considered unacceptable
+by the sig-auth community ([see the Slack
+thread](https://kubernetes.slack.com/?redir=%2Farchives%2FC0EN96KUY%2Fp1582313541102300%3Fthread_ts%3D1582308815.101400%26cid%3DC0EN96KUY)).
+
+### RPC vs exec
+
+Similarly as in case of [external credential
+providers](../541-external-credential-providers), external plugin could be
+exposed as a network endpoint. Instead of executing a binary and passing
+request/response over arguments/`stdout`, client could open a network connection
+and send request/response over that.
+
+The downsides of this approach compared to exec model are:
+
+- if external signer is remote, design for client authentication is required
+  (aka "chicken-and-egg problem"),
```

Comment:
This problem has to be solved for any remote signer, regardless of local kubeconfig API.
But I agree that this is a concern for the auth provider, not for `client-go`